### DISCUSSION TITLE
Liveserver

### DISCUSSION THREAD

**Discussion by gldtn** [zed-industries/zed#6649] [Feature Requests]
I don't see any editors with out of the box the ability to communicate with Liveserver web browser extension w/o installing additional plugins. Do you guys plan to implementing your own or will you guys be leaving it for when plugins are possible? Or do you guys have something else in mind?

IMO the ability to view your code live on save is a must have tool for any editor!

**Comment by gldtn** [comment]
For those like me, not running a framework that have tools like these built-in and want to have "live"(reloads on save) preview in your browser. You can install [browsersync](https://browsersync.io/)

This assumes you have NPM installed..


Install:
```
npm install -g browser-sync
```

create `bs-config.js` file inside your project root dir with:
_this config assumes your running your own php server(like [Herd](https://herd.laravel.com/)), see the [Browsersync documentation](https://browsersync.io/docs)!_
```
module.exports = {
    files: ["./**/*.{html,php,css,js}"],
    proxy: {
        target: "http://domain.test",
        ws: true,
    },
    ui: false,
};
```

Go into your project root path and run:
```
 browser-sync start --config bs-config.js
```

Unlike Liverserver, you don't need a browser plugin for this to work!

**Reply by SinaMombeiny** [reply]
thanks
but still a Liverserver extension would be nice

**Reply by damian-robinson** [reply]
> thanks but still a Liverserver extension would be nice

here, here! ⬆️ built-in is better

**Reply by Jacobinoo** [reply]
> thanks but still a Liverserver extension would be nice

definitely!!

**Comment by yhabib** [comment]
For what it's worth, I've installed `live-server` and created the following task:

```
  {
    "label": "Live Server",
    "command": "live-server",
    "use_new_terminal": true,
    "reveal": "never"
  }
```

It works like the VS Code plugin! :)

**Reply by SolaKabuta** [reply]
You made my day !

**Reply by killitlv** [reply]
new to zed cod editor installed(windows). live server don't shows any option to serve live. where should i put the following task code

[Your Project Root]/.zed/tasks.json ?
{
    "label": "Live Server",
    "command": "live-server",
    "use_new_terminal": true,
    "reveal": "never"
  }

is this json file works ?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
